sdk/keyvault/Azure.ResourceManager.KeyVault/tests/Extensions.cs

```diff
@@ -75,7 +75,7 @@ public static bool IsEqual(this DeletedVaultResource deletedVault, VaultData cre
             Assert.AreEqual(createdVault.Name, deletedVault.Data.Name);
             Assert.AreEqual(createdVault.Id, deletedVault.Data.Properties.VaultId);
             Assert.AreEqual("Microsoft.KeyVault/deletedVaults", deletedVault.Data.ResourceType);
-            Assert.True(createdVault.Tags.DictionaryEqual(deletedVault.Data.Properties.Tags));
+            Assert.True(createdVault.Tags.DictionaryEqual(deletedVault.Data.Properties.Tags.ToDictionary(kvp => kvp.Key, kvp => kvp.Value)));
```

Comment:
why do we need this?

Reply:
deletedVault.Data.Properties.Tags is `IReadOnlyDictionary` but `createdVault.Tags` is IDictionary`